src/macros/macrorecorder.h

```diff
@@ -54,6 +53,17 @@ class MacroRecorder : public RecordingManagerBase {
     FRIEND_TEST(MacroRecordingTest, ClaimRecording);
     FRIEND_TEST(MacroRecordingTest, RecordCueJump);
 
+    enum class State : uint8_t {
+        /// Nothing is going on
+        Disabled,
+        /// Recording is active, but nothing currently going on
+        Armed,
+        /// An Action is currently being recorded
+        Recording,
+    };
+
+    State getState() const;
+
```

Comment:
This is super confusing. It looks redundant, but it isn't because the values of the `State` enum mean something different from the `Status` enum. Please do not implement a custom poor-mans locking mechanism. Remove this and use a proper SPSC queue instead.

Reply:
Any progress on this?

Reply:
Why/How should I use a SPSCQueue here? What am I supposed to enqueue?

Reply:
Right now you're using a C array for storing macro actions and track the recording state like this:
- Armed = Recording active not recording an action
- Recording = Recording active and currently recording an action

I assume this is necessary in case the user stops recording while an action is currently written to the array. It's possible that the macro action has been written only partially. Is that correct?

If so, the underlying issue is that storing the macro action is not "atomic". This can be solved by using a SPSCQueue instead of a C-Array for storing the macro actions. Then you don't need the separate state tracking nor reset the recording state after every write anymore. When the recording is stopped, you can then just read all recorded actions from the queue.

Reply:
Ah! Now that makes a lot of sense :)